src/filefingerprint.cpp

```diff
@@ -190,7 +190,7 @@ bool FileFingerprint::genfingerprint(FileAccess* fa, bool ignoremtime)
                 if (!fa->frawread(block, sizeof block,
                                   (size - sizeof block)
                                   * (i * blocks + j)
-                                  / (sizeof crc / sizeof *crc * blocks - 1)))
+                                  / (sizeof crc / sizeof *crc * blocks - 1), i || j, true))
                 {
                     size = -1;
                     return true;
```

Comment:
```suggestion
                    fa->closef();
                    return true;
```

Reply:
actually this should only be done in case the failure does not come from fopen. We could add a isOpened()  and do `if (fa->isOpened()) fa->closef();`

Reply:
I saw in your PR for this branch, and that the file is always already opened before calling this function fingerprinting!   Great spotting, I should have seen that too.  Thanks very much for the fix :-)